procinterrupts.c

```diff
@@ -143,6 +143,36 @@ static void guess_arm_irq_hints(char *name, struct irq_info *info)
 	
 	
 }
+
+static int is_arm_irq_aff_cannot_change(int irq)
+{
+	char buf[PATH_MAX] = { 0 };
+	FILE *file = NULL;
+	char *line = NULL;
+	size_t size = 0;
+	int ret = 0;
+
+	snprintf(buf, PATH_MAX - 1, "/proc/irq/%i/smp_affinity", irq);
+	file = fopen(buf, "r+");
+	if (!file)
+		return -1;
+
+	if (getline(&line, &size, file) <= 0) {
+		ret = -1;
+		goto out;
+	}
+
```

Comment:
How does this even work?  I don't see how being unable to read the smp_affinity of an irq results in deciding that its unchangeable.  If the smp_affinity is unchangeable, it will either sink the write operation and succeed, but not change the contents, or the file will have no writable permissions.  Either way, a failure to read shouldn't have any bearing on its ability to have its affinity set